Task 1: How many rows and columns are there in the table?

In [1]:
import pandas as pd
data = pd.read_csv("pandas_task.csv")
data.shape

(15, 5)

Task 2: Print row 3 to 8 using loc or iloc

In [4]:
data = pd.read_csv("pandas_task.csv")
print(data.iloc[3:9])


  Destinations  Feedback score (out of 10)  Average hotel star rating  \
3      Germany                         8.4                        3.8   
4           UK                         8.2                        3.2   
5          USA                         5.1                        3.0   
6       Canada                         9.7                        4.5   
7       Mexico                         7.8                        1.8   
8    Argentina                         8.0                        3.7   

   Number of all-inclusive hotels Most visited city  
3                             156            Munich  
4                              98            London  
5                             234          New York  
6                             412           Toronto  
7                             567            Cancun  
8                             231      Buenos Aires  


Task 3: Find the mean of all-inclusive hotels across all destinations

In [5]:
data["Number of all-inclusive hotels"].mean()

215.2

Task 4: Find the lowest-scoring destination

In [20]:
df = pd.DataFrame(data)
#Finding the min value of the feedback score column:
df["Feedback score (out of 10)"].min()
#Finding the index:
df["Feedback score (out of 10)"].idxmin()
#Using the index to find the location:
df.iloc[10]


Destinations                      Austria
Feedback score (out of 10)            3.8
Average hotel star rating             3.5
Number of all-inclusive hotels        123
Most visited city                  Vienna
Name: 10, dtype: object

Task 5: Find the highest scoring destination

In [23]:
df["Feedback score (out of 10)"].max()
df["Feedback score (out of 10)"].idxmax()
#Using the index 6 to find the location
df.iloc[6]

Destinations                       Canada
Feedback score (out of 10)            9.7
Average hotel star rating             4.5
Number of all-inclusive hotels        412
Most visited city                 Toronto
Name: 6, dtype: object

Task 6: Find all the destinations where there are more than 9 all-inclusive hotels

In [28]:
myfilter = data["Number of all-inclusive hotels"] > 9
myfilter
all_inc = data[myfilter]
all_inc

Destinations  Feedback score (out of 10)  Average hotel star rating  \
0           France                         8.5                        4.1   
1          Belgium                         7.2                        3.7   
2           Russia                         5.6                        2.6   
3          Germany                         8.4                        3.8   
4               UK                         8.2                        3.2   
5              USA                         5.1                        3.0   
6           Canada                         9.7                        4.5   
7           Mexico                         7.8                        1.8   
8        Argentina                         8.0                        3.7   
9      Switzerland                         9.2                        4.9   
10         Austria                         3.8                        3.5   
11           Italy                         7.9                        3.3   
12           Spain                         9.0                        2.8   
13  Czech Republic                         5.8                        3.3   
14       Australia                         9.1                        3.5   

    Number of all-inclusive hotels Most visited city  
0                              226             Paris  
1                               12          Brussels  
2                              312     St Petersburg  
3                              156            Munich  
4                               98            London  
5                              234          New York  
6                              412           Toronto  
7                              567            Cancun  
8                              231      Buenos Aires  
9                               39            Zurich  
10                             123            Vienna  
11                             187              Rome  
12                             436         Barcelona  
13                              18            Prague  
14                             177         Melbourne

Task 7: Filter the data by score above 8

In [30]:
above_eight = data["Feedback score (out of 10)"] > 8
superscore = data[above_eight]
superscore
 

Destinations  Feedback score (out of 10)  Average hotel star rating  \
0        France                         8.5                        4.1   
3       Germany                         8.4                        3.8   
4            UK                         8.2                        3.2   
6        Canada                         9.7                        4.5   
9   Switzerland                         9.2                        4.9   
12        Spain                         9.0                        2.8   
14    Australia                         9.1                        3.5   

    Number of all-inclusive hotels Most visited city  
0                              226             Paris  
3                              156            Munich  
4                               98            London  
6                              412           Toronto  
9                               39            Zurich  
12                             436         Barcelona  
14                             177         Melbourne

Task 8: Filter the data by score below 2

In [31]:
below_two = data["Feedback score (out of 10)"] < 2
trouble = data[below_two]
trouble

Empty DataFrame
Columns: [Destinations, Feedback score (out of 10), Average hotel star rating, Number of all-inclusive hotels, Most visited city]
Index: []

Task 9: Is there a correlation between the number of all-inclusive hotels and the feedback score?I

In [32]:
df["Number of all-inclusive hotels"].corr(df["Feedback score (out of 10)"])

0.23038745315357206

The correlation coefficient is 0.23 (to 2 s.f.). Therefore the correlation is weak and therefore negligible.

Task 10: Draw a diagram to visualise destination and highest score